User/InMemoryUserRepository.php

```diff
@@ -42,20 +42,16 @@ public function supportsClass($class)
     /**
      * {@inheritdoc}
      */
-    public function add(User $user, $save = true)
+    public function add(User $user)
     {
-        $this->users[] = $user;
+        $this->save($user);
     }
-
+    
     /**
      * {@inheritdoc}
      */
-    public function save(User $user = null)
+    public function save(User $user)
     {
-        if ($user === null) {
-            return;
-        }
-
```

Comment:
volgens mij moet bij in memory de add `$this->users[] = $user;` bevatten en de save niets? (omdat objecten by reference worden bijgehouden)? Dat lijkt me logischer? Nu kan je door een user twee keer te saven hem twee keer toevoegen.